packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -11580,28 +11576,51 @@ export class AdapterClass extends EventEmitter {
 
                         this.oStates = _states;
                         this.subscribeStates('*');
+
                         if (this._firstConnection) {
                             this._firstConnection = false;
-                            typeof this._options.ready === 'function' && this._options.ready();
-                            this.emit('ready');
+                            this._callReadyHandler();
                         } else {
-                            typeof this._options.reconnect === 'function' && this._options.reconnect();
+                            // TODO: is this really needed? If yes, we should document it.
+                            if (typeof this._options.reconnect === 'function') {
+                                this._options.reconnect();
+                            }
                             this.emit('reconnect');
                         }
                         this.adapterReady = true;
                     });
                 } else if (!this._stopInProgress) {
-                    typeof this._options.ready === 'function' && this._options.ready();
-                    this.emit('ready');
+                    this._callReadyHandler();
                     this.adapterReady = true;
 
-                    // todo remove it later, when the error is fixed
+                    // todo remove it later, when the error is fixed. Which error, if someone knows explain please!
```

Comment:
@GermanBluefox You remember? Else I would remove it because not heared of any logging error since ages

Reply:
@GermanBluefox @Apollon77  I have removed it now, if someone has concerns we can readd